builder/build_validate.go

```diff
@@ -592,6 +597,35 @@ func (bundleDiff *mcaBundleDiff) diffBundles(bundle string, fromInfo, toInfo map
 	return nil
 }
 
+// getSkippedFiles returns a map of files to skip for the MCA check. rpm 4.12 used
+// md5 file hashes by default and rpm 4.14 switched the default to sha256. As a result,
+// in build 31680, rpms started to use sha256 hashes instead of md5. Since MCA assumes
+// that hash changes indicate content changes, rpm comparisons with different hash types
+// are not possible without expanding the scope of MCA to verify content hashes.
+func getSkippedFiles(fromFiles, toFiles map[string]*fileInfo) map[string]bool {
+	skipMap := make(map[string]bool)
+	for fName, toFile := range toFiles {
+		// The os-release file is expected to change every build, so a change caused
+		// by a different hashing algorithm will not cause an MCA error.
+		if fName == "/usr/lib/os-release" {
+			continue
+		}
+
+		// Flag comparisons between md5 hashes (len == 32) and sha256 hashes (len == 64),
+		// so they can be skipped later.
+		fromFile := fromFiles[fName]
+		if fromFile != nil {
+			if (fromFile.hashLen == 32 || fromFile.hashLen == 64) &&
+				(toFile.hashLen == 32 || toFile.hashLen == 64) &&
+				fromFile.hashLen != toFile.hashLen {
```

Comment:
Any reason why we have these 3 checks:
```
			if (fromFile.hashLen == 32 || fromFile.hashLen == 64) &&
				(toFile.hashLen == 32 || toFile.hashLen == 64) &&
				fromFile.hashLen != toFile.hashLen {
```

instead of just
`			if fromFile.hashLen != toFile.hashLen {`

Reply:
Symlinks have a hash length of 0, so the following hash length combinations are possible and should not be skipped: 0->32, 0->64, 32->0, 64->0

I can readjust the logic to better handle symlinks while also dealing with future hashing algorithm changes.

Reply:
I think this is good for now. Maybe just add the comment about symlinks the code for readability.